framework/base/src/types/interaction/result_handlers/token_properties_result.rs

```diff
@@ -0,0 +1,81 @@
+use multiversx_sc_codec::{
+    DecodeErrorHandler, TopDecodeInput, TopDecodeMulti, TopDecodeMultiInput,
+};
+
+const MAX_BUFFER_SIZE: usize = 62;
+
+#[derive(Debug, Clone, Default)]
+pub struct TokenPropertiesResult {
+    pub num_decimals: usize,
+    pub is_paused: bool,
+    pub can_upgrade: bool,
+    pub can_mint: bool,
+    pub can_burn: bool,
+    pub can_change_owner: bool,
+    pub can_pause: bool,
+    pub can_freeze: bool,
+    pub can_wipe: bool,
+    pub can_add_special_roles: bool,
+    pub can_transfer_nft_create_role: bool,
+    pub nft_create_stopped: bool,
+    pub num_wiped: usize,
+}
+
+impl TokenPropertiesResult {
+    fn fetch_struct_field(&mut self, input: &[u8]) {
+        let is_true = |value: &[u8]| value == b"true";
+
+        if let Some(value) = input.strip_prefix(b"NumDecimals-") {
```

Comment:
Put all prefixes in a const slice and iterate over them (except the last one). This will lead to smaller contracts and less duplicated code.